gitlab.sh

```diff
@@ -21,8 +21,57 @@
 # 2016/05/18 http://stackoverflow.com/questions/59895/can-a-bash-script-tell-what-directory-its-stored-in
 DIR_THIS="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
+#######################################
+# This script is needed to handle ssh key on gitlab CI,
+# and assumes the steps in the tutorial https://docs.gitlab.com/ee/ci/ssh_keys/README.html is followed.
+#
+# Globals:
+#   (None)
+# Arguments:
+#   _SSH_PRIVATE_KEY: Content of SSH private key that is defined on gitlab as a "secret variable". See detail at https://docs.gitlab.com/ee/ci/ssh_keys/README.html
+# Returns:
+#   (None)
+#######################################
+inject_sshkey () {
+    set -x
+  
+    if [ -z "$_SSH_PRIVATE_KEY" ]; then
+        echo "SSH_PRIVATE_KEY is not found.";
+        return 1;
+    fi      
+    
+    # Install ssh-agent if not already installed, it is required by Docker.
+    # (change apt-get to yum if you use a CentOS-based image)
+    which ssh-agent || ( apt-get update -y && apt-get install openssh-client -y )
```

Comment:
This depends on the executor and the images used for the executor.
For our default example it should be something like this: `apk --no-cache add openssh-client`.